File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/strategy/transport/Plane.java
package com.bigboxcode.designpattern.strategy.transport;

import com.bigboxcode.Transport;

public class Plane implements Transport {

    @Override
    public void start() {
        System.out.println("Plane started");
    }

    @Override
    public void stop() {
        System.out.println("Plane stopped");
    }

    @Override
    public void repair() {
        System.out.println("Plane repair");
    }

    @Override
    public void getInfo() {
        System.out.println("Transport type: Plane");
        System.out.println("Number of wheels: 3");
        System.out.println("Average Weight: 50,000 Pounds");
    }

    @Override
    public void operate() {
        System.out.println("Flying plane ............");
    }
}

File: design-patterns/php/strategy/transport/Plane.php
<?php
// Plane.php

namespace BigBoxCode\DesignPattern\Strategy\Transport;


class Plane implements Transport {
    public function start(): void {
        echo "Plane started\n";
    }

    public function stop(): void {
        echo "Plane stopped\n";
    }

    public function repair(): void {
        echo "Plane repair\n";
    }

    public function getInfo(): void {
        echo "Transport type: Plane\n";
        echo "Number of wheels: 3\n";
        echo "Average Weight: 50,000 Pounds\n";
    }

    public function operate(): void {
        echo "Flying plane ............\n";
    }
}
